sdk/src/device_utils.h

```diff
@@ -32,14 +32,27 @@
 #ifndef DEVICE_UTILS_H
 #define DEVICE_UTILS_H
 
+#include <assert.h>
 #include <cstddef>
+#include <glog/logging.h>
 #include <inttypes.h>
 
 namespace aditof {
 static void deinterleave(const char *source, uint16_t *destination,
                          size_t source_len, size_t dest_width,
                          size_t dest_height) {
 
+    if (source == nullptr) {
+        LOG(ERROR) << "Received source null pointer";
+        return;
+    }
+
+    if (destination == nullptr) {
+        LOG(ERROR) << "Received destination null pointer";
+        return;
+    }
+    assert(source_len > 0);
```

Comment:
`source_len` has the type `size_t` which by definition is unsigned, so this is not needed.

Reply:
In this way we will prevent the length to be set to 0.